experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.ps1

```diff
@@ -0,0 +1,27 @@
+param (
+    [string]$LOC = "./"  # Sets a parameter named $LOC with a default value of "./"
+)
+
+Write-Host "hello"  # Displays "hello" on the console
+Write-Host $LOC  # Displays the value of $LOC on the console
+
+$secretsDirectory = Join-Path $LOC "secrets"  # Combines $LOC and "secrets" to create the secrets directory path
+New-Item -ItemType Directory -Path $secretsDirectory -Force | Out-Null  # Creates the secrets directory if it doesn't exist
+
+$privateKeyPath = Join-Path $secretsDirectory "jenkins_agent_ed"  # Combines the secrets directory path with the private key filename
+$publicKeyPath = Join-Path $secretsDirectory "jenkins_agent_ed.pub"  # Combines the secrets directory path with the public key filename
+
+# Remove existing keys
+Remove-Item $privateKeyPath, $publicKeyPath -ErrorAction SilentlyContinue  # Removes any existing private and public key files
+
+# Generate new ed25519 SSH key pair
+ssh-keygen -t ed25519 -f $privateKeyPath -P "" -N ""  # Generates a new ed25519 SSH key pair with an empty passphrase
```

Comment:
I think some considerations we need to add to the tutorial.

We should add the prerequisite. As far as I know, Windows user should install the ssh-keygen separately, which mean they should have this before running this script.

I have a suggestion to check all the prerequisites and show them which tools that should be installed before running the scripts.

Reply:
Thank you @berviantoleo .
I happen to have `ssh-keygen` directly available within Powershell, but that's maybe it's a recent version.

Reply:
Ah, okay. I just found [an article](https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement#user-key-generation).

We can assume the ssh-keygen is already there.